sources/packages/backend/libs/utilities/src/system-configurations-constants.ts

```diff
@@ -30,6 +30,17 @@ export const ORM_CACHE_LIFETIME = 10 * 60 * 1000;
  */
 export const ORM_CACHE_REDIS_COMMAND_TIMEOUT = 5 * 1000;
 
+/**
+ * Maximum number of open connections that
+ * are allowed for connection pool.
+ */
+export const CONNECTION_POOL_MAX_CONNECTIONS = 20;
```

Comment:
Should we match this with the number used by the `processInParallel` method?
Wonder that by default the `processInParallel` will use 32 as a baseline depending on the POD cores which means that 32 parallel operations will be started and there will be 20 connections available meaning that 12 parallel operations will be waiting for a connection to be acquired from the pool (that now has 2min timeout).
It does not seem a big deal under regular operation but it can be pretty impactful if the DB is under some pressure or taking too long to respond.

Reply:
Thanks for bringing this point. As discussed we can take this up on upcoming dev sync up.